src/Sarif/Warnings.cs

```diff
@@ -10,8 +10,9 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class Warnings
     {
-        public const string Wrn997InvalidTarget = "WRN997.InvalidTarget";
-        public const string Wrn998NotSupportedPlatform = "WRN998.NotSupportedPlatform";
+        public const string Wrn997InvalidTarget        = "WRN997";
+        public const string Wrn998NotSupportedPlatform = "WRN998";
+        public const string Wrn999RuleDisabled         = "WRN999";
```

Comment:
So there's a convention here and in Errors.cs that WRN998.(explanation)/ERR998.(Explanation) are used for rule-disabling errors/warnings. 

Similarly, (WRN/ERR)997 are configuration/target errors (missing pdb, invalid target, cannot write log, etc.).

I think that model is a little better than just addressing these sequentially as we assign them.

Reply:
warning WRN999 : Rule 'BA2010' was explicitly disabled by the user.
error ERR997.ExceptionLoadingPdb : BA2006 : 'MixedMode_x86_VS2013_MissingPdb.dll' was not evaluated for check 'BuildWithSecureTools'.

See output above for why I made the change. I think I should further qualify the ids as you say, then maybe we can consider changing the console output to strip the friendly sub-name (since the output text makes things clear).


---
In reply to: [169829092](https://github.com/Microsoft/sarif-sdk/pull/791#discussion_r169829092) [](ancestors = 169829092)

Reply:
Makes sense.  My main thinking is that we should be consistent.

Maybe WRN998.RuleDisabled:  '(x)' was explicitly disabled by the user and WRN.998.RuleDisabled: '(x)' cannot execute on this platform?